File: java/src/main/java/com/company/automation/models/RegisterRequest.java
package com.company.automation.models;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Payload for {@code POST /users/register}.
 *
 * <p>{@code toString} is overridden because this record carries a password. A record's generated
 * one prints every component, which would put the password into any log line or assertion message
 * that touched it.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record RegisterRequest(
    @JsonProperty("first_name") String firstName,
    @JsonProperty("last_name") String lastName,
    @JsonProperty("address") PostalAddress address,
    @JsonProperty("phone") String phone,
    @JsonProperty("dob") String dateOfBirth,
    @JsonProperty("email") String email,
    @JsonProperty("password") String password) {

  @Override
  public String toString() {
    return "RegisterRequest[email=" + email + ", password=***REDACTED***]";
  }

  /** A postal address, as the registration endpoint expects it. */
  public record PostalAddress(
      @JsonProperty("street") String street,
      @JsonProperty("city") String city,
      @JsonProperty("state") String state,
      @JsonProperty("country") String country,
      @JsonProperty("postal_code") String postalCode) {}
}

File: csharp/src/AutomationFramework.Core/Models/RegisterRequest.cs
using System.Text.Json.Serialization;
using AutomationFramework.Core.Support;

namespace AutomationFramework.Core.Models;

/// <summary>
/// Payload for <c>POST /users/register</c>.
/// </summary>
/// <remarks>
/// <c>ToString</c> is overridden because this record carries a password.
/// </remarks>
public sealed record RegisterRequest(
    [property: JsonPropertyName("first_name")] string FirstName,
    [property: JsonPropertyName("last_name")] string LastName,
    [property: JsonPropertyName("address")] PostalAddress Address,
    [property: JsonPropertyName("phone")] string Phone,
    [property: JsonPropertyName("dob")] string DateOfBirth,
    [property: JsonPropertyName("email")] string Email,
    [property: JsonPropertyName("password")] string Password)
{
    public override string ToString()
        => $"RegisterRequest {{ Email = {Email}, Password = {Redaction.Marker} }}";
}

/// <summary>A postal address, as the registration endpoint expects it.</summary>
public sealed record PostalAddress(
    [property: JsonPropertyName("street")] string Street,
    [property: JsonPropertyName("city")] string City,
    [property: JsonPropertyName("state")] string State,
    [property: JsonPropertyName("country")] string Country,
    [property: JsonPropertyName("postal_code")] string PostalCode);
